Caboodle/Magnetometer/Magnetometer.ios.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using CoreMotion;
+using Foundation;
+
+namespace Microsoft.Caboodle
+{
+    public static partial class Magnetometer
+    { // Timing intervales to match android sensor speeds in seconds
+        // https://stackoverflow.com/questions/10044158/android-sensors
+        internal const double FastestInterval = .02;
+        internal const double GameInterval = .04;
+        internal const double UiInterval = .08;
+        internal const double NormalInterval = .225;
+
+        internal static bool IsSupported =>
+            Platform.MotionManager?.MagnetometerAvailable ?? false;
+
+        internal static void PlatformStart(SensorSpeed sensorSpeed)
+        {
+            var manager = Platform.MotionManager;
+            switch (sensorSpeed)
+            {
+                case SensorSpeed.Fastest:
+                    manager.MagnetometerUpdateInterval = FastestInterval;
+                    break;
+                case SensorSpeed.Game:
+                    manager.MagnetometerUpdateInterval = GameInterval;
+                    break;
+                case SensorSpeed.Normal:
+                    manager.MagnetometerUpdateInterval = NormalInterval;
+                    break;
+                case SensorSpeed.Ui:
+                    manager.MagnetometerUpdateInterval = UiInterval;
+                    break;
+            }
+
+            manager.StartMagnetometerUpdates(NSOperationQueue.CurrentQueue, DataUpdated);
+
+            void DataUpdated(CMMagnetometerData data, NSError error)
```

Comment:
Maybe move this out into the class to be consistent - unless it has to be here?

Reply:
#resolved